test/Npgsql.PluginTests/NodaTimeTests.cs

```diff
@@ -795,6 +795,16 @@ public async Task Normalize_period_on_write()
         Assert.That(dbValue, Is.EqualTo(expectedAfterRoundtrip));
     }
 
+    [Test]
+    public async Task Period_write_throw_on_overflow()
+    {
+        var periodBuilder = new PeriodBuilder
+        {
+            Years = int.MaxValue
+        };
+        await AssertTypeUnsupportedWrite<Period, OverflowException>(periodBuilder.Build(), "interval");
```

Comment:
Ideally, rather than a cryptic OverflowException we'd throw an ArgumentException (I think) with a clear message, saying that the provided Period is out of range for the PG interval type.

Reply:
Hmm, I suppose we can just put `checked` in try/catch and handle it there.

Reply:
Related to this I hope we'll get non throwing overflow checking operations at some point https://github.com/dotnet/runtime/issues/13026